core/src/main/scala/com/spotify/featran/transformers/HashNHotWeightedEncoder.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.featran.transformers
+
+import java.util
+
+import com.spotify.featran.FeatureBuilder
+import com.twitter.algebird.HLL
+
+import scala.collection.SortedMap
+import scala.collection.mutable.{Map => MMap}
+
+object HashNHotWeightedEncoder {
+  /**
+   * Transform a collection of weighted categorical features to columns of weight sums, with at
+   * most N values.
+   *
+   * Weights of the same labels in a row are summed instead of 1.0 as is the case with the normal
+   * [[NHotEncoder]].
+   *
+   * Missing values are transformed to [0.0, 0.0, ...].
+   *
+   * When using aggregated feature summary from a previous session, unseen labels are ignored.
+   */
+  def apply(name: String, hashBucketSize: Int = 0)
+  : Transformer[Seq[WeightedLabel], HLL, Int] =
+    new HashNHotWeightedEncoder(name, hashBucketSize)
+}
+
+private class HashNHotWeightedEncoder(name: String, hashBucketSize: Int)
+  extends BaseHashHotEncoder[Seq[WeightedLabel]](name, hashBucketSize) {
+
+  override def prepare(a: Seq[WeightedLabel]): HLL =
+    a.map(_.name).map(hllMonoid.toHLL(_)).reduce(hllMonoid.plus)
+
+  override def buildFeatures(a: Option[Seq[WeightedLabel]],
+                             c: Int,
+                             fb: FeatureBuilder[_]): Unit = {
+    fb.init(c)
+    a match {
+      case Some(xs) => {
+        val weights = MMap.empty[String, Double].withDefaultValue(0.0)
```

Comment:
Should be a mutable SortedMap of `[Int, Double]` instead so strings that collides can have their weights summed up?